src/controllers/chat-controllers/get-chat-token-controller.ts

```diff
@@ -0,0 +1,42 @@
+import {Options, ResponseType} from "../../services/http-service";
+import {ErrorResponse} from "../../api/types";
+import {Indexed} from "../../core/types";
+import {ChatTokenResponse, GetChatTokenAPI} from "../../api/chat-api/get-chat-token-api";
+
+
+const getChatTokenAPI = new GetChatTokenAPI();
+
+export class GetChatTokenController {
+  static async get(chatId: number): Promise<string | void> {
+    try {
+      // Запускаем крутилку
+      return getChatTokenAPI.get(getOptions(), chatId)
+        .then((response: ChatTokenResponse | ErrorResponse) => {
+          // Останавливаем крутилку
+          if (isErrorResponse(response)) {
+            throw new Error(response.reason);
+          }
+
+          return response?.token;
+        })
+        .catch((error) => {
+          console.error(error, chatId);
+          // Останавливаем крутилку
+        })
+    } catch (error) {
+      console.error(error, chatId);
+      // Логика обработки ошибок
+    }
+  }
+}
+
+function isErrorResponse(response: Indexed): response is ErrorResponse {
+  return !!response?.reason;
```

Comment:
Часто наглядней пользоваться явным приведением: 
```suggestion
  return Boolean(response?.reason);
```

Как показывает практика, большую часть времени мы код читаем, а не пишем. Так что наглядность - достаточно важный параметр)